### PR TITLE
backend/fix/bpp-searchReq-location-mapping-update

### PR DESCRIPTION
**Pull request by Shikhar-24**
## Type of Change

- [x] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description

driver side location mapping update for searchRequest

### CODE REVIEW

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Beckn/Update.hs [hunk 1]
```diff
@@ -158,6 +160,12 @@ handler (UEditLocationReq EditLocationReq {..}) = do
     QLM.create pickupMapForBooking
     pickupMapForRide <- SLM.buildPickUpLocationMapping startLocation.id rideId.getId DLM.RIDE (Just person.merchantId) (Just person.merchantOperatingCityId)
     QLM.create pickupMapForRide
+    searchRequest <- SQSR.findByTransactionId transactionId
```
Comment (jaypalmudaliyar24) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Beckn/Update.hs:163]: `findByTransactionIdAndMerchantId` use this query instead.

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Beckn/Update.hs [hunk 2]
```diff
@@ -158,6 +160,9 @@ handler (UEditLocationReq EditLocationReq {..}) = do
     QLM.create pickupMapForBooking
     pickupMapForRide <- SLM.buildPickUpLocationMapping startLocation.id rideId.getId DLM.RIDE (Just person.merchantId) (Just person.merchantOperatingCityId)
     QLM.create pickupMapForRide
+    searchReq <- SQSR.findByTransactionIdAndMerchantId transactionId person.merchantId >>= fromMaybeM (InvalidRequest ("Search request not found for the transactionId " <> transactionId))
```
Comment (jaypalmudaliyar24) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Beckn/Update.hs:163]: Use this error instead, `SearchRequestDoesNotExist` or `SearchRequestNotFound`.

**Review by jaypalmudaliyar24 — DISMISSED** [review]

**Review by jaypalmudaliyar24 — APPROVED** [review]

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Beckn/Update.hs [hunk 3]
```diff
@@ -91,7 +92,8 @@ data EditLocationReq = EditLocationReq
     origin :: Maybe DL.Location,
     destination :: Maybe DL.Location,
     status :: Enums.OrderStatus,
-    bapBookingUpdateRequestId :: Text
+    bapBookingUpdateRequestId :: Text,
+    transactionId :: Text
```
Comment (khuzema786) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Beckn/Update.hs:96]: will this be backward compatible as older versions would not be sending transactionId and it's non maybe.
Make this `Maybe` if this is currently used in Prod.

**Review by khuzema786 — APPROVED** [review]

**Review by khuzema786 — CHANGES_REQUESTED** [review]
Change needed

**Review by khuzema786 — APPROVED** [review]

**Pull request merged**